### PR TITLE
Create CONTRIBUTING.md from README `Get involved with the community` section

### PR DESCRIPTION
**Pull request by benoitdm-oslandia**
This PR creates the `CONTRIBUTING.md` file from the README `Get involved with the community` section. 

This makes the project more 'compliant' with other OS projects and best practices. It also improves the access to the developer guidelines at <https://docs.qgis.org>.

Linked to https://github.com/qgis/QGIS-Documentation/pull/9229

### CODE REVIEW

**Comment by benoitdm-oslandia** [comment]
The test failure is not linked to any modifications made in this PR...

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Contributing
+
+## Developer guide
+
+ A [complete developer guide](https://docs.qgis.org/latest/en/docs/developers_guide/index.html) is also available to explain the build process, the unit testing system and more!
```
Comment (DelazJ) [inline CONTRIBUTING.md:5]: The [install.md](https://github.com/qgis/QGIS/blob/master/INSTALL.md) file provides good references for build process and is worth a mention IMHO.

Comment (benoitdm-oslandia) [reply CONTRIBUTING.md]: I changes the 1st section, does it suite you?

**Review by nyalldawson — APPROVED** [review]

**Comment by nyalldawson** [comment]
Great move 👍

**Comment by agiudiceandrea** [comment]
Have you seen https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTING.md

**Comment by benoitdm-oslandia** [comment]
@agiudiceandrea arrgg no! :(

**Comment by benoitdm-oslandia** [comment]
@agiudiceandrea I have created a double `contributing.md`.  But the old `.github/contributing.md` is only referenced by the `.github/pull_request_template.md`. 

I would like to adapt the `.github/pull_request_template.md` to reference the <https://docs.qgis.org/latest/en/docs/developers_guide/index.html> instead and move the `.github/contributing.md` part about the `prepare_commit.sh` into the `https://docs.qgis.org/3.34/en/docs/developers_guide/git.html#procedure` between point 4 and 5. 

In the `.github/pull_request_template.md` the `prepare_commit.sh` part could be suppressed.

At final, `.github/contributing.md` could be then suppressed.

(cc @nyalldawson)

**Comment by nyalldawson** [comment]
@benoitdm-oslandia sounds good to me!

**Comment by benoitdm-oslandia** [comment]
@agiudiceandrea gentle ping!

**Pull request merged**